File: app/src/main/java/org/wikipedia/settings/SiteInfo.java
package org.wikipedia.settings;

import android.support.annotation.Nullable;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class SiteInfo {
    @SuppressWarnings("unused") @Nullable private String mainpage;
    @SuppressWarnings("unused") @Nullable private String sitename;
    @SuppressWarnings("unused") @Nullable private String lang;
    @SuppressWarnings("unused") @Nullable private List<LanguageVariants> variants;
    @SuppressWarnings("unused") @SerializedName("readinglists-config")
    @Nullable private ReadingListsConfig readingListsConfig;

    @Nullable public String mainPage() {
        return mainpage;
    }

    @Nullable public ReadingListsConfig readingListsConfig() {
        return readingListsConfig;
    }

    @Nullable public String lang() {
        return lang;
    }

    public boolean hasVariants() {
        return variants != null && variants.size() > 0;
    }

    public static class ReadingListsConfig {
        @SuppressWarnings("unused") private int maxListsPerUser;
        @SuppressWarnings("unused") private int maxEntriesPerList;
        @SuppressWarnings("unused") private int deletedRetentionDays;

        public int maxEntriesPerList() {
            return maxEntriesPerList;
        }
    }

    private static class LanguageVariants {
        @SuppressWarnings("unused") private String code;
        @SuppressWarnings("unused") private String name;
    }
}

File: app/src/main/java/org/wikipedia/settings/SiteInfo.kt
package org.wikipedia.settings

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
class SiteInfo(
    val lang: String? = null,
    @SerialName("readinglists-config") val readingListsConfig: ReadingListsConfig? = null
) {

    // TODO: remove this class when temporary accounts are rolled out to all wikis.
    @Serializable
    class AutoCreateTempUser(val enabled: Boolean = true)

    @Serializable
    class ReadingListsConfig(val maxListsPerUser: Int = 0,
                             val maxEntriesPerList: Int = 0,
                             val deletedRetentionDays: Int = 0)
}
